hr_expense_tax_adjustment/models/hr_expense.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023-today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class HrExpense(models.Model):
+
+    _inherit = "hr.expense"
+
+    tax_line_ids = fields.One2many(
+        "hr.expense.tax",
+        "expense_id",
+        "Detailed Taxes",
+        readonly=True,
+        copy=True,
+        states={
+            "draft": [("readonly", False)],
+            "reported": [("readonly", False)],
+            "refused": [("readonly", False)],
+            "approved": [("readonly", False)],
+        },
+    )
+
+    @api.onchange(
+        "product_id", "quantity", "unit_amount", "tax_ids", "company_id", "currency_id"
+    )
+    def _onchange_amount_setup_tax_lines(self):
+        if self.quantity and self.unit_amount and self.tax_ids:
+            self._setup_tax_lines()
+        else:
+            self.tax_line_ids = self.env["hr.expense.tax"]
+
+    def _setup_tax_lines(self):
+        """Setup the taxes on the expense."""
+        self.tax_line_ids = self.env["hr.expense.tax"]
+
+        currency = self.currency_id or self.company_id.currency_id
+        taxes = self.tax_ids.with_context(round=True).compute_all(
```

Comment:
IMO we can use directly self.tax_ids without compute_all 
self.tax_ids is computed in V16 so we don't need to call compute_all

https://github.com/odoo/odoo/blob/16.0/addons/hr_expense/models/hr_expense.py#L67
https://github.com/odoo/odoo/blob/16.0/addons/hr_expense/models/hr_expense.py#L205